test/Generator/TypeGeneratorTest.php

```diff
@@ -80,6 +80,87 @@ public function testRejectsInvalidTypeString(string $typeString)
     }
 
     /**
+     * @dataProvider validTypeArrayProvider
+     *
+     * @param string $typeArray
+     * @param string $expectedReturnType
+     */
+    public function testFromValidStringAlias(array $typeArray, string $expectedReturnType)
+    {
+        $generator = TypeGenerator::fromTypeArray($typeArray);
```

Comment:
I'd make this `TypeGenerator::fromAliasedType(string $type)` (the scalar type hint can't be added tho: needs to be checked manually)